lib/find-all.js

```diff
@@ -0,0 +1,37 @@
+var isntDesignDoc = require('./utils/isnt-design-doc')
+
+module.exports = findAll
+
+/**
+ * finds all existing objects in local database.
+ *
+ * @param  {String}   prefix     optional id prefix
+ * @param  {Function} [filter]   Function returning `true` for any object
+ *                               to be returned.
+ * @return {Promise}
+ */
+
+function findAll (state, prefix, filter) {
+  var options = {
+    include_docs: true
+  }
+
+  if (prefix) {
+    options.startkey = prefix
+    options.endkey = prefix + '\uffff'
```

Comment:
it’s the highest Unicode character, so it makes sure it gives us all keys which start with the prefix string, see http://docs.couchdb.org/en/2.0.0/couchapp/views/collation.html#string-ranges (I don’t know why they use `ufff0` instead of `uffff` in the example though)